### PR TITLE
Clean up PlrDirOK, PlrClrTrans and PlrDoTrans

### PR DESCRIPTION
**Pull request by AJenbo**
Bin exact

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -1003,135 +1003,53 @@ BOOL __fastcall SolidLoc(int x, int y)
...
+
+	if ( px < 0 || !dPiece[px][py] || !PosOkPlayer(pnum, px, py) ) {
+		return FALSE;
 	}
-	if ( v9 )
-		return 1;
-	return 0;
+
+	BOOL isOk = TRUE;
+	if ( dir == 6 ) {
```
Comment (squidcc) [inline Source/player.cpp]: Does dir hold a value based on the direction enum?

File: Source/player.cpp [hunk 2]
```diff
@@ -1003,135 +1003,53 @@ BOOL __fastcall SolidLoc(int x, int y)
...
-	int v7; // ebp
-	bool result; // eax
-	bool v9; // zf
-	int p; // [esp+10h] [ebp-4h]
-
-	v2 = pnum;
-	v3 = dir;
-	p = pnum;
-	if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( (DWORD)pnum >= MAX_PLRS ) {
```
Comment (squidcc) [inline Source/player.cpp]: Why not change the function parameter to DWORD?

Comment (nomdenom) [reply Source/player.cpp]: The original code generally had `(DWORD)` casts from `pnum` which we can see from the assert message AFAIK.

Comment (squidcc) [reply Source/player.cpp]: From which build?

Comment (seritools) [reply Source/player.cpp]: These assert strings are included in the 1.00 Debug release hidden inside the DIABDAT.mpq, see https://github.com/diasurgical/devil-nightly#introduction (remove the first 36 bytes from the exe after unpacking to get a proper PE executable)

**Pull request merged**